test/jdk/javax/swing/JComponent/7154030/bug7154030.java

```diff
@@ -86,14 +89,19 @@ public void run() {
 
                     frame.setContentPane(desktop);
                     frame.setSize(300, 300);
-                    frame.setLocation(0, 0);
+                    frame.setLocationRelativeTo(null);
                     frame.setVisible(true);
                     frame.setDefaultCloseOperation(JFrame.DISPOSE_ON_CLOSE);
                 }
             });
 
-            robot.waitForIdle(500);
-            imageInit = robot.createScreenCapture(new Rectangle(0, 0, 300, 300));
```

Comment:
The usage of size 300x300 has the same issues as locx/locy before. It is part of the window bounds.

Reply:
but 300x300 is set in frame setSize..if I use frame getBounds instead of already set 300x300, that will not be right, according to me...What if getBounds() has some bug (in some platform) and return some other width/height (say 1 pixel less, we might have similar bug in 8196465) that what is set in setSize, then the test might pass even though robot is capturing wrong bounds.

Reply:
> but 300x300 is set in frame setSize..if I use frame getBounds instead of already set 300x300, that will not be right, according to me...

The test should set the size 300,300 to the frame and then request and use the real bounds of the frame.

> What if getBounds() has some bug (in some platform) and return some other width/height (say 1 pixel less, we might have similar bug in 8196465) that what is set in setSize, then the test might pass even though robot is capturing wrong bounds.

The getBounds() simply returns the size which was set by the native system, so is it a bug in getBounds() or the real bounds of the frame "say 1 pixel less"?

Reply:
Any further feedback?